.circleci/config.yml

```diff
@@ -6,11 +6,16 @@ jobs:
       - checkout
 
       - run:
-        name: Build
-        command: |
-          docker build -t mozilla/remote-settings-uptake-health:$CIRCLE_BRANCH .
+          name: Build
+          command: |
+            docker build -t mozilla/remote-settings-uptake-health:$CIRCLE_SHA1 .
```

Comment:
Hey @glasserc Can you take a look at this?

[The documentation](https://circleci.com/docs/2.0/building-docker-images/#example) for using the built-in `docker` in a CircleCI machine is to use:

```
- run: docker build -t company/app:$CIRCLE_BRANCH .
```
But that won't work because the value of `$CIRCLE_BRANCH` is something like `pull/9` (from it being the ID=9 github pull request) and you can't take it with 2 `/` characters. I.e.

```
- run: docker build -t company/app:pull/9 .
```
will fail because the tag name has two `/` in it. 

According to this: https://discuss.circleci.com/t/circle-branch-content-has-changed/10552/2
the `$CIRCLE_BRANCH` *used* to be the branch name. E.g. `my-feature-branch`. 

So, at first I thought I'll just remove the `$CIRCLE_BRANCH` part and do:

```
docker build -t mozilla/remote-settings-uptake-health
```
...which would work. But, it makes me nervous that if I make PR there might be confusing and conflicting caching done by CircleCI. 

I *think* the `$CIRCLE_SHA1` is essentially the git commit (if the repo is git) so it's guaranteed to always be relevantly differerent for each build. 

What do you think?

(N.B. I'm not super concerned since this repo is so relatively simple. But it can't hurt to learn.)

Reply:
I'm not really an expert on Docker or CircleCI...

Why do we need to tag the Docker image here? It doesn't seem like we push the image. What's the value of giving it a tag which is a SHA?

Is this intended to run on every pull request, or every merge, or what?

Reply:
Yeah, it's the intention to run these lines for every PR  but also when we land things in `master`. What's not in this file yet is that if the `$CIRCLE_BRANCH  == 'master' || $CIRCLE_TAG` (<-- pseudo-code) then it'll ship to dockerhub. A la https://github.com/mozilla/normandy/blob/8d00e6cdea3ad2021486bbb6b9eb53fffe534cf1/.circleci/config.yml#L95-L100

Perhaps tagging it has nothing do with whether docker caches anything. 

Also, can you do `docker build ...` and `docker run ...` *without* a tag??

Reply:
I guess it's hard to automatically guess the name of the container for the `docker run` command unless you do `docker build` with a tag..

I think `docker build` doesn't do any caching. I think the issue I encountered that one time was due to `docker-compose` using `docker run` without doing a `docker pull`. `docker run` will re-used tagged containers if available. I'm not sure if it will conflict with other branches happening at the same time?

You could also probably use a `sed` command to replace the `/` in `$CIRCLE_BRANCH`.